### PR TITLE
adds delay to carding AI

### PR DESCRIPTION
**Pull request by toomykins**
[tweak]
Screenshot_2020-08-02_18-52-08
saw pic related and agreed
intellicards shouldn't be a metastrat for dealing with ~~malf~~ rogue AI's. ~~Malfs~~ Rogue AI's should be destroyed, and the cards used to fix em. (IMO)
gives AI time to flash carder and call a borg. 

:cl:
 * tweak: intellicards now have a delay.

### CODE REVIEW

File: code/game/machinery/computer/ai_core.dm [hunk 1]
```diff
@@ -150,7 +150,10 @@
 
 /obj/structure/AIcore/deactivated/attackby(var/obj/item/device/aicard/A as obj, var/mob/user as mob)
 	if(istype(A, /obj/item/device/aicard))//Is it?
-		A.transfer_ai("INACTIVE","AICARD",src,user)
+		if(do_after(user, src, 40))
+			if(!istype(src) && !istype(A))
```
Comment (DamianX) [inline code/game/machinery/computer/ai_core.dm]: these two checks will never fail

File: code/game/objects/items/devices/aicard.dm [hunk 2]
```diff
@@ -16,15 +16,17 @@
 		return ..()
 	if(M.mind && M.mind.current != M)
 		return ..()
+	if(do_after(user, M, 40))
+		if(!istype(src) && !istype(M))
```
Comment (DamianX) [inline code/game/objects/items/devices/aicard.dm]: these two checks will never fail

**Comment by DamianX** [comment]
This doesn't change anything, go for it.

File: code/game/machinery/computer/ai_core.dm (hunk 1 above)
Comment (toomykins) [reply code/game/machinery/computer/ai_core.dm]: don't know if true but i'll trust you on it

**Comment by BomberBro** [comment]
Unless something changed very recently, malf AIs already can't be carded, did you test this?

**Comment by toomykins** [comment]
alright i admit i didn't know you couldn't card malf AIs, I did test this with normal AIs (and just now with malf)
I guess it still applies to rogue lawset AI's.

**Comment by TGFH** [comment]
I don't see why this would be desirable at all. Neverplayer in the thread didn't know that malf AIs can't be carded anyway and AIs that have been subverted would just get assraped by this change because the meta would to be toolbox them and then forget that they ever existed.

**Comment by ShiftyRail** [comment]
MalfAIs can't be carded after delta. They can still be picked up during the APC Hacking phase.

**Pull request closed without merging**